### PR TITLE
New/codelists

### PR DESCRIPTION
**Pull request by JimBryant7**
Added 9 extra codes to the admin-geography code list
2 new code lists for the tax benefit dataset
2 new code lists for the regional gdp datasets
added 1 new code to type-of-prices code list
1 new geography code list -> nuts-geography
added corresponding .grm & .yaml files

requires a sanity check on the codes and labels within the code lists

### CODE REVIEW

File: code-list-scripts/gremlin/remaining-life-expectancy.grm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '5-years-remaining-life-expectancy').
+  fold().coalesce(unfold(), addV('_code').
+    property(single, 'listID', 'remaining-life-expectancy').property(single, 'value', '5-years-remaining-life-expectancy')
+  ).next()
+g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '5-years-remaining-life-expectancy').outE().fold().coalesce(unfold(),
+  g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '5-years-remaining-life-expectancy').
+    addE('usedBy').to(
+      g.V().hasLabel('_code_list').has('listID', 'remaining-life-expectancy').has('edition', 'one-off')
+    ).property(single, 'label', ' % with 5 years remaining LE')).next()
```
Comment (daiLlew) [inline code-list-scripts/gremlin/remaining-life-expectancy.grm]: Whitespace before `%`

File: code-list-scripts/gremlin/remaining-life-expectancy.grm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '15-years-remaining-life-expectancy').
+  fold().coalesce(unfold(), addV('_code').
+    property(single, 'listID', 'remaining-life-expectancy').property(single, 'value', '15-years-remaining-life-expectancy')
+  ).next()
+g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '15-years-remaining-life-expectancy').outE().fold().coalesce(unfold(),
+  g.V().hasLabel('_code').has('listID', 'remaining-life-expectancy').has('value', '15-years-remaining-life-expectancy').
+    addE('usedBy').to(
+      g.V().hasLabel('_code_list').has('listID', 'remaining-life-expectancy').has('edition', 'one-off')
+    ).property(single, 'label', '% with 15 years remaining LE')).next()
```
Comment (daiLlew) [inline code-list-scripts/gremlin/remaining-life-expectancy.grm]: No whitespace before `%`.

File: code-list-scripts/gremlin/remaining-life-expectancy.grm (hunk 1 above)
Comment (daiLlew) [reply code-list-scripts/gremlin/remaining-life-expectancy.grm]: Consistency. Also consider expanding `LE` to life expectancy.

File: geo-specific-codelist/nuts-geography.cypher [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+MERGE (node:`_code`:`_code_geography` { value:"UKL" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKL" }) MERGE (node)-[:usedBy { label:"Wales"}]->(parent);
+MERGE (node:`_code`:`_code_geography` { value:"UKM" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKM" }) MERGE (node)-[:usedBy { label:"Scotland"}]->(parent);
+MERGE (node:`_code`:`_code_geography` { value:"UKN" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKN" }) MERGE (node)-[:usedBy { label:"Northern Ireland"}]->(parent);
+MERGE (node:`_code`:`_code_geography` { value:"T" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"T" }) MERGE (node)-[:usedBy { label:"Total"}]->(parent);
+MERGE (node:`_code`:`_code_geography` { value:"UKZ" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKZ" }) MERGE (node)-[:usedBy { label:"Extra-regio"}]->(parent);
```
Comment (daiLlew) [inline geo-specific-codelist/nuts-geography.cypher]: `n` missing

File: geo-specific-codelist/nuts-geography.cypher [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+MERGE (node:`_code`:`_code_geography` { value:"UKZZ" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKZZ" }) MERGE (node)-[:usedBy { label:"Extra-regio"}]->(parent);
+MERGE (node:`_code`:`_code_geography` { value:"UKZZZ" });
+MATCH (parent:`_code_list`:`_code_list_nuts-geography`),(node:`_code`:`_code_geography` { value:"UKZZZ" }) MERGE (node)-[:usedBy { label:"Extra-regio"}]->(parent);
```
Comment (daiLlew) [inline geo-specific-codelist/nuts-geography.cypher]: 3 labels are the same and the values look suspicious :shipit: ?

File: geo-specific-codelist/gremlin/nuts-geography.grm [hunk 5]
```diff
@@ -0,0 +1,2606 @@
...
+
+
+g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKJ34').
+  fold().coalesce(unfold(), addV('_code').
+    property(single, 'listID', 'geography').property(single, 'value', 'UKJ34')
+  ).next()
+g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKJ34').outE().fold().coalesce(unfold(),
+  g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKJ34').
+    addE('usedBy').to(
+      g.V().hasLabel('_code_
```
Comment (daiLlew) [inline geo-specific-codelist/gremlin/nuts-geography.grm]: Make sure this file is also regenerated (see comments for corresponding cypher script)

File: geo-specific-codelist/nuts-geography.cypher (hunk 3 above)
Comment (JimBryant7) [reply geo-specific-codelist/nuts-geography.cypher]: I also thought this originally but "extra-regio" is the correct name

File: geo-specific-codelist/nuts-geography.cypher (hunk 4 above)
Comment (JimBryant7) [reply geo-specific-codelist/nuts-geography.cypher]: these codes do look a bit odd but these codes are used in the nuts geography as it goes down to three levels. But I've removed UKZZ & UKZZZ, just too confusing

File: geo-specific-codelist/gremlin/nuts-geography.grm [hunk 6]
```diff
@@ -0,0 +1,2584 @@
...
```
Comment (daiLlew) [inline geo-specific-codelist/gremlin/nuts-geography.grm]: I googled this to make sure it was a real thing... who knew.

File: geo-specific-codelist/gremlin/nuts-geography.grm [hunk 7]
```diff
@@ -0,0 +1,2584 @@
...
+
+g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKE11').
+  fold().coalesce(unfold(), addV('_code').
+    property(single, 'listID', 'geography').property(single, 'value', 'UKE11')
+  ).next()
+g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKE11').outE().fold().coalesce(unfold(),
+  g.V().hasLabel('_code').has('listID', 'geography').has('value', 'UKE11').
+    addE('usedBy').to(
+      g.V().hasLabel('_code_list').has('listID', 'nuts-geography').has('edition', 'one-off')
+    ).property(single, 'label', 'Kingston upon Hull, City of')).next()
```
[the remark this answers is not in the document]
Comment (JimBryant7) [reply geo-specific-codelist/gremlin/nuts-geography.grm]: it is correct, some of these geography names are just peculiar..

File: code-list-scripts/gremlin/remaining-life-expectancy.grm (hunk 1 above)
Comment (daiLlew) [reply code-list-scripts/gremlin/remaining-life-expectancy.grm]: Still leading whitespace on this one.

**Pull request merged**